app/lib/features/home/pages/dashboard.dart

```diff
@@ -33,7 +33,9 @@ class _DashboardState extends ConsumerState<Dashboard> {
   @override
   void initState() {
     super.initState();
-    _checkIfSpacesPresent();
+    WidgetsBinding.instance.addPostFrameCallback((timestamp) {
+      _checkIfSpacesPresent();
```

Comment:
can you explain why we would delay this operation?

Reply:
This doesn't necessary delays the operation, it's more like a safety approach to not altering any state of UI while widget is already building. This call would ensure we are running state code after widget is properly initialised. As I see we're using ```ref.watch()``` which is reactive to put changes in UI. Removing this would lead to exception, as I was getting when navigating to dashboard.

Reply:
```final spaces = await ref.watch(spacesProvider.future);```

Reply:
but the called function (internally) is async, this delaying doesn't "ensure" that unfortunately.

Yeah... ref-read and ref-watch in initState are problematic ... we might want to find some guidelines for that, too ...